services/permissions/database.go

```diff
@@ -95,6 +95,7 @@ func (db *Database) GetPermissions() (rst []DappPermissions, err error) {
 		}
 		dapps[perms.Name] = &perms
 	}
+	rows.Close()
```

Comment:
I think you want to close it just after you check for `err`, it might return early and not close

Reply:
`defer` ftw!